parquet-column/src/main/java/org/apache/parquet/column/values/bloom/Bloom.java

```diff
@@ -0,0 +1,485 @@
+
+  // Algorithm applied of this bloom filter.
+  public ALGORITHM bloomFilterAlgorithm = ALGORITHM.BLOCK;
+
+  // A hash set to cache the distinct value from column.
+  private HashSet<Long> elements = new HashSet<Long>();
+
+  // The underlying byte array for bloom filter bitset.
+  private byte[] bitset;
+
+  // The size of bitset in byte.
+  private int numBytes;
+
+  // List of byte input to construct the bloom filter.
+  private final List<BytesInput> inputs= new ArrayList<>(4);
+
+  /**
+   * Constructor of bloom filter, if numBytes is zero, bloom filter bitset
+   * will be created lazily and the number of bytes will be calculated through
+   * distinct values in cache.
+   * @param numBytes The number of bytes for bloom filter bitset, set to zero can
+   *                 let it calculate number automatically by using default FPP.
+   * @param hash The hash strategy bloom filter apply.
+   * @param algorithm The algorithm of bloom filter.
+   */
+  public Bloom(int numBytes, HASH hash, ALGORITHM algorithm) {
+    if (numBytes != 0) {
+      initBitset(numBytes);
+    }
+    this.bloomFilterHash = hash;
+    this.bloomFilterAlgorithm = algorithm;
+  }
+
+  /**
+   * Construct the bloom filter with given bit set.
+   * @param bitset The given bitset to construct bloom filter.
+   * @param hash The hash strategy bloom filter apply.
+   * @param algorithm The algorithm of bloom filter.
+   */
+  public Bloom(byte[] bitset, HASH hash, ALGORITHM algorithm) {
+    this.bitset = bitset;
+    this.numBytes = bitset.length;
+    this.bloomFilterHash = hash;
+    this.bloomFilterAlgorithm = algorithm;
+  }
+
+  /**
+   * Construct the bloom filter with given bit set.
+   * @param input The given bitset to construct bloom filter.
+   * @param hash The hash strategy bloom filter apply.
+   * @param algorithm The algorithm of bloom filter.
+   */
+  public Bloom(ByteBuffer input, HASH hash, ALGORITHM algorithm) {
+    if (input.isDirect()){
+      bitset = new byte[input.remaining()];
+      input.get(bitset);
+    } else {
+      this.bitset = input.array();
+    }
+    this.numBytes = input.array().length;
+    this.bloomFilterHash = hash;
+    this.bloomFilterAlgorithm = algorithm;
+  }
+
+  /**
+   * Create a new bitset for bloom filter, at least 256 bits will be create.
+   * @param numBytes number of bytes for bit set.
+   */
+  private void initBitset(int numBytes) {
+    if (numBytes < MINIMUM_BLOOM_SIZE) {
+      numBytes = MINIMUM_BLOOM_SIZE;
+    }
+
+    if (numBytes > ParquetProperties.DEFAULT_MAXIMUM_BLOOM_FILTER_SIZE) {
+      numBytes = ParquetProperties.DEFAULT_MAXIMUM_BLOOM_FILTER_SIZE;
+    }
+
+    // One bucket alignment.
+    numBytes = (numBytes + BYTES_PER_BUCKET - 1) & ~(BYTES_PER_BUCKET - 1);
+
+    ByteBuffer bytes = ByteBuffer.allocate(numBytes);
+    this.bitset = bytes.array();
+    this.numBytes = numBytes;
+  }
+
+  /**
+   * @return Bytes in buffered in this bloom filter.
+   */
+  public BytesInput getBytes() {
```

Comment:
I don't see any callers.

Reply:
this is used when serialize bloom filter to file.